testkit/jvm/src/main/scala/scalaz/zio/testkit/TestScheduler.scala

```diff
@@ -0,0 +1,80 @@
+package scalaz.zio.testkit
+
+import scalaz.zio._
+import scalaz.zio.clock._
+import scalaz.zio.duration.{ Duration, _ }
+import scalaz.zio.internal.Scheduler.CancelToken
+import scalaz.zio.internal.{ Scheduler => IScheduler }
+import scalaz.zio.scheduler.Scheduler
+import scalaz.zio.testkit.TestScheduler._
+
+/**
+ * Implementation of Scheduler.Service for testing. The passed Ref[TestClock.Data] will be used to determine when
+ * to run the scheduled runnables. Make sure to call shutdown() to force execution of all remaining tasks.
+ */
+final case class TestScheduler(ref: Ref[TestClock.Data], runtime: Runtime[Clock]) extends Scheduler.Service[Any] {
+
+  private[this] val ConstFalse = () => false
+
+  override def scheduler: ZIO[Any, Nothing, IScheduler] =
```

Comment:
The one thing I recommend we do before merging is add a top-level `shutdown: UIO[Unit]` function to this service, to make it easier to shut it down from tests.

Reply:
Would something like this do?
` 
trait TestIScheduler extends IScheduler {
  def safeShutdown(): UIO[Unit]
}
`